source/ShipInfoDisplay.cpp

```diff
@@ -303,6 +318,27 @@ void ShipInfoDisplay::UpdateAttributes(const Ship &ship, const Depreciation &dep
 	heatTable.push_back(Format::Number(60. * emptyMass * .1 * attributes.Get("heat dissipation")));
 	// Pad by 10 pixels on the top and bottom.
 	attributesHeight += 30;
+	
+	// Flight check warnings
+	if(ship.IsYours())
+	{
+		vector<string> warnings = ship.FlightCheck();
```

Comment:
Why make FlightCheck() a Ship class member, rather than a ShipInfoDisplay member? The Ship class is pretty large already, and the game only uses FlightCheck() for informational purposes - the rest of Ship class members are used to instantiate ships or help them make decisions in flight (e.g. used by the Ship, AI, and Engine classes).

Reply:
Because it will be used in other places (this is a work in progress) and it just makes sense.
The Ship class has methods that provide information about the ship. I consider this function to be the same kind of function and the consequences of this information is left to outside code.

Reply:
I suppose if you are planning to reference this data from PlayerInfoPanel (which ships have warnings), PlanetPanel/PlayerInfo (can we take off?) or OutfitterPanel/ShopPanel (can we leave the outfitter?), then it being a Ship class member makes sense.